pakyow-core/spec/unit/middleware/www_enforcer_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'spec_helper'
+require 'pakyow/core/middleware/www_enforcer'
+
+describe Pakyow::Middleware::WWWEnforcer do
+  let :app do
+    double
+  end
+
+  let :instance do
+    Pakyow::Middleware::WWWEnforcer.new(app)
+  end
+
+  let :env do
+    {
+      'SERVER_NAME' => host,
+      'REQUEST_METHOD' => method
+    }
+  end
+
+  let :host do
+    'pakyow.com'
+  end
+
+  let :host_www do
+    'www.pakyow.com'
+  end
+
+  let :host_subdomain do
+    'foo.pakyow.com'
+  end
+
+  let :method do
+    'GET'
+  end
+
+  before do
+    allow(app).to receive(:enforce_www) { true }
+    allow(Pakyow).to receive(:app).and_return(app)
+    allow(app).to receive(:call)
+  end
+
+  describe '#call' do
```

Comment:
Here's an example of what I was thinking: https://github.com/pakyow/pakyow/commit/a641285906b95e96d421cbb077ae9ae47c1c463f